packages/perseus-editor/src/__stories__/perseus-frame-component.tsx

```diff
@@ -0,0 +1,184 @@
+import {
+    ServerItemRenderer,
+    HintRenderer,
+    PerseusI18nContextProvider,
+    type SharedRendererProps,
+} from "@khanacademy/perseus";
+import * as React from "react";
+
+import {testDependenciesV2} from "../../../../testing/test-dependencies";
+import {mockStrings} from "../../../perseus/src/strings";
+
+import type {PerseusItem, Hint} from "@khanacademy/perseus-core";
+
+type PerseusFrameData =
+    | {
+          type: "question";
+          data: SharedRendererProps & {
+              item: PerseusItem;
+              initialHintsVisible?: number;
+              device?: string;
+              reviewMode?: boolean;
+              legacyPerseusLint?: any;
+          };
+      }
+    | {
+          type: "hint";
+          data: SharedRendererProps & {
+              hint: Hint;
+              pos?: number;
+              bold?: boolean;
+          };
+      };
+
+/**
+ * PerseusFrameComponent is a component that renders Perseus content in an iframe.
+ * It is used to render the content of the EditorPage in order to provide previews
+ * of the questions and hints.
+ *
+ * @returns A PerseusFrameComponent that renders the content of the EditorPage.
+ */
+const PerseusFrameComponent = () => {
```

Comment:
This is fantastic that you reverse-engineered this info. Ideally, I'd love to extract the same component out of webapp and move it into Perseus (so that we'd have both sides of the communication channel in Perseus) and then eventually refactor to use the same component in the stories as in production. Future work, but this definitely paves the way! Thanks.